includes/payment-methods/class-klarna-payment-method.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Class Affirm_Payment_Method
+ *
+ * @package WCPay\Payment_Methods
+ */
+
+namespace WCPay\Payment_Methods;
+
+use WC_Payments_Token_Service;
+use WC_Payments_Utils;
+use WCPay\MultiCurrency\MultiCurrency;
+
+/**
+ * Affirm Payment Method class extending UPE base class
+ */
+class Klarna_Payment_Method extends UPE_Payment_Method {
+
+	const PAYMENT_METHOD_STRIPE_ID = 'klarna';
+
+	/**
+	 * Constructor for link payment method
+	 *
+	 * @param WC_Payments_Token_Service $token_service Token class instance.
+	 */
+	public function __construct( $token_service ) {
+		parent::__construct( $token_service );
+		$this->stripe_id                    = self::PAYMENT_METHOD_STRIPE_ID;
+		$this->title                        = __( 'Klarna', 'woocommerce-payments' );
+		$this->is_reusable                  = false;
+		$this->icon_url                     = plugins_url( 'assets/images/payment-methods/klarna.svg', WCPAY_PLUGIN_FILE );
+		$this->currencies                   = [ 'AUD', 'CAD', 'CHF', 'CZK', 'DKK', 'EUR', 'GBP', 'NOK', 'NZD', 'PLN', 'SEK', 'USD' ];
+		$this->accept_only_domestic_payment = true;
+		$this->countries                    = [ 'AU', 'AT', 'BE', 'CA', 'CZ', 'DK', 'FI', 'FR', 'GR', 'DE', 'IE', 'IT', 'NL', 'NZ', 'NO', 'PL', 'PT', 'ES', 'SE', 'CH', 'GB', 'US' ];
```

Comment:
Countries need to be confirmed yet - we don't know which of them  has fees  finalized yet.

Reply:
Yes, I've noted that to myself in the description of PR as well.